### PR TITLE
doc: Add comment for fetchurl for name & url

### PR DESCRIPTION
**Pull request by fzakaria**
# Motivation
fetchurl can be given a name and url aside from just the url. Giving a name can be useful if the url has invalid characters such as tilde for the store.

image

# Context
#10532

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

**Comment by fzakaria** [comment]
What are the steps to preview the doc ?
I can include a screenshot of how it rendered.

File: src/libexpr/primops/fetchTree.cc [hunk 1]
```diff
@@ -529,9 +529,20 @@ static void prim_fetchurl(EvalState & state, const PosIdx pos, Value * * args, V
...
-      Download the specified URL and return the path of the downloaded file.
+      If args is a URL, return the path of the downloaded file.
+      Otherwise, it can be an attribute with the following attributes
+      (all except url are optional):
+
+      - `url`
+
+        The URL of the file to download.
+
+      - `name` (default: `url without the protocol`)
```
Comment (edolstra) [inline src/libexpr/primops/fetchTree.cc:542]:
```suggestion
      - `name` (default: the last path component of the URL)
```
or something like that.

File: src/libexpr/primops/fetchTree.cc [hunk 2]
```diff
@@ -529,9 +529,20 @@ static void prim_fetchurl(EvalState & state, const PosIdx pos, Value * * args, V
 
 static RegisterPrimOp primop_fetchurl({
     .name = "__fetchurl",
-    .args = {"url"},
+    .args = {"args"},
     .doc = R"(
-      Download the specified URL and return the path of the downloaded file.
+      If args is a URL, return the path of the downloaded file.
+      Otherwise, it can be an attribute with the following attributes
+      (all except url are optional):
```
Comment (edolstra) [inline src/libexpr/primops/fetchTree.cc:536]:
```suggestion
      Download the specified URL and return the path of the downloaded file.
      `arg` can be either a string denoting the URL, or an attribute set with the following attributes:
```

File: src/libexpr/primops/fetchTree.cc [hunk 3]
```diff
@@ -529,9 +529,20 @@ static void prim_fetchurl(EvalState & state, const PosIdx pos, Value * * args, V
 
 static RegisterPrimOp primop_fetchurl({
     .name = "__fetchurl",
-    .args = {"url"},
+    .args = {"args"},
```
Comment (edolstra) [inline src/libexpr/primops/fetchTree.cc:532]:
```suggestion
    .args = {"arg"},
```

**Review by edolstra — CHANGES_REQUESTED** [review]

**Review by edolstra — APPROVED** [review]

**Pull request merged**